vumi/persist/tests/test_fields.py

```diff
@@ -3,15 +3,117 @@
 """Tests for vumi.persist.fields."""
 
 from datetime import datetime
+from functools import wraps
 
-from vumi.message import Message
+from twisted.internet.defer import inlineCallbacks, returnValue
 
+from vumi.message import Message, TransportUserMessage
 from vumi.persist.fields import (
     ValidationError, Field, Integer, Unicode, Tag, Timestamp, Json,
-    ListOf, SetOf, Dynamic, FieldWithSubtype, Boolean, VumiMessage)
-from vumi.tests.helpers import VumiTestCase
+    ListOf, SetOf, Dynamic, FieldWithSubtype, Boolean, VumiMessage,
+    ForeignKey, ManyToMany)
+from vumi.persist.model import Manager, Model
+from vumi.tests.helpers import VumiTestCase, MessageHelper, import_skip
 
 
+def needs_riak(method):
+    """
+    Mark a test method as needing Riak setup.
+    """
+    method.needs_riak = True
+    return method
+
+
+class ModelFieldTestsDecorator(object):
+    """
+    Class decorator for replacing `@needs_riak`-marked test methods with two
+    wrapped versions, one for each Riak manager.
+
+    This is used here instead of the more usual mechanism of a mixin and two
+    subclasses because we have a lot of small test classes which have several
+    test methods that don't need Riak.
```

Comment:
Hmm. I think I'll retract this since it's worth something to not have to remember to create subclasses of each test.